In [2]:
#IMPORTS AND LOADING DATA
%matplotlib inline
import pandas as pd
import numpy as np
import datetime 
import matplotlib.pyplot as plt

def configure_plots():
    '''Configures plots by making some quality of life adjustments'''
    for _ in range(2):
        plt.rcParams['figure.figsize'] = [16/2, 9/2]
        plt.rcParams['axes.titlesize'] = 20
        plt.rcParams['axes.labelsize'] = 16
        plt.rcParams['xtick.labelsize'] = 14
        plt.rcParams['ytick.labelsize'] = 14
        plt.rcParams['lines.linewidth'] = 2

configure_plots()

med = pd.read_csv("../data/med.csv")
people = pd.read_csv("../data/people.csv")
combined= pd.read_csv("../data/combined.csv")

med23 = med[med["Date"] == str(datetime.date(2023, 4, 1))]
combined23 = combined[combined["Date"] == str(datetime.date(2023, 4, 1))]
people23 = people[people["Individual Id"].isin(med23["Individual Id"].to_list())]

In [37]:
print(f'Number of people at fair: {people23.shape[0]}')

percent_male = np.round(np.mean(combined23["Gender"]=="m")*100, 2)
print(f'Percent men: {percent_male}%')
print(f'Percent women: {100-percent_male}%')

percent_kids = np.round(np.mean(combined23["Age At Fair"]<18)*100, 2)

print(f'Percent kids (under 18): {percent_kids}%')
print(f'Percent adults: {100-percent_kids}%')

mean_age = np.round(np.mean(combined23["Age At Fair"]),2)
print(f'Average age: {mean_age}')

list = combined23["Family Id"].to_list() 
counts = {x:list.count(x) for x in list}

families = []
for id in np.unique(list): 
    if counts[id] > 1: 
        families.append(id)

print(f'Number of families (with at least two people): {len(families)}')

adult = combined23[combined23["Age At Fair"]>17]

mean_bmi_adult = np.round(np.mean(adult["Bmi"]),2)
print(f'Average bmi among adults: {mean_bmi_adult}')
obese = adult[adult["Bmi"]>=30].shape[0]
overweight = adult[(adult["Bmi"]>=25) & (adult["Bmi"]<30)].shape[0]
healthy = adult[(adult["Bmi"]>=18.5) & (adult["Bmi"]<25)].shape[0]
underweight = adult[adult["Bmi"]<18.5].shape[0]
bmi_count = adult.dropna(subset="Bmi").shape[0]
print(f'Percent of adults obese: {np.round(obese/bmi_count*100,2)}%')
print(f'Percent of adults overweight: {np.round(overweight/bmi_count*100,2)}%')
print(f'Percent of adults healthy: {np.round(healthy/bmi_count*100,2)}%')
print(f'Percent of adults underweight: {np.round(underweight/bmi_count*100,2)}%')

mean_sugar = np.round(np.mean(combined23["Soda Sugar"]),2)
print(f'Average sugar: {mean_sugar}')

mean_blood_sugar = np.round(np.mean(combined23[combined23["Fast"]==True]["Blood Sugar"]),2)
print(f'Average blood sugar (among fasting): {mean_blood_sugar}')

mean_cav = np.round(np.mean(combined23["Cavities"]),2)
print(f'Average cavities: {mean_cav}')



Number of people at fair: 182
Percent men: 36.26%
Percent women: 63.74%
Percent kids (under 18): 57.69%
Percent adults: 42.31%
Average age: 20.5
Number of families (with at least two people): 54
Average bmi among adults: 30.26
Percent of adults obese: 49.28%
Percent of adults overweight: 27.54%
Percent of adults healthy: 18.84%
Percent of adults underweight: 4.35%
Average sugar: 35.95
Average blood sugar (among fasting): 106.9
Average cavities: 2.44


In [40]:
for date in np.sort(combined["Date"].unique()): 
    print(f'{date} Number of people at fair: {combined[combined["Date"]==date].shape[0]}')

2021-05-01 Number of people at fair: 110
2021-11-01 Number of people at fair: 212
2022-04-01 Number of people at fair: 180
2022-11-01 Number of people at fair: 152
2023-04-01 Number of people at fair: 182


In [42]:
for date in np.sort(combined["Date"].unique()): 
    percent_male = np.round(np.mean(combined[combined["Date"]==date]["Gender"]=="m")*100, 2)
    print(f'{date} Percent men: {percent_male}%')
    print(f'{date} Percent women: {100-percent_male}%')

2021-05-01 Percent men: 39.09%
2021-05-01 Percent women: 60.91%
2021-11-01 Percent men: 38.68%
2021-11-01 Percent women: 61.32%
2022-04-01 Percent men: 39.44%
2022-04-01 Percent women: 60.56%
2022-11-01 Percent men: 40.79%
2022-11-01 Percent women: 59.21%
2023-04-01 Percent men: 36.26%
2023-04-01 Percent women: 63.74%


In [44]:
for date in np.sort(combined["Date"].unique()): 
    percent_kids = np.round(np.mean(combined[combined["Date"]==date]["Age At Fair"]<18)*100, 2)
    print(f'{date} Percent kids (under 18): {percent_kids}%')
    print(f'{date} Percent adults: {100-percent_kids}%')

2021-05-01 Percent kids (under 18): 58.18%
2021-05-01 Percent adults: 41.82%
2021-11-01 Percent kids (under 18): 53.3%
2021-11-01 Percent adults: 46.7%
2022-04-01 Percent kids (under 18): 53.33%
2022-04-01 Percent adults: 46.67%
2022-11-01 Percent kids (under 18): 61.18%
2022-11-01 Percent adults: 38.82%
2023-04-01 Percent kids (under 18): 57.69%
2023-04-01 Percent adults: 42.31%


In [54]:
for date in np.sort(combined["Date"].unique()): 
    list = combined[combined["Date"]==date]["Family Id"].to_list() 
    counts = {x:list.count(x) for x in list}

    families = []
    for id in np.unique(list): 
        if counts[id] > 2: 
            families.append(id)

    print(f'Number of families (with at least two people): {len(families)}')

Number of families (with at least two people): 20
Number of families (with at least two people): 37
Number of families (with at least two people): 37
Number of families (with at least two people): 35
Number of families (with at least two people): 41


In [57]:
for date in np.sort(combined["Date"].unique()): 
    mean_age = np.round(np.mean(combined[combined["Date"]==date]["Age At Fair"]),2)
    print(f'Average age: {mean_age}')

Average age: 18.66
Average age: 21.04
Average age: 21.58
Average age: 20.29
Average age: 20.5


Number of people at fair: 182
Percent men: 36.26%
Percent women: 63.74%
Percent kids (under 18): 57.69%
Percent adults: 42.31%
Average age: 20.5
Number of families (with at least two people): 54
Average bmi among adults: 30.26
Percent of adults obese: 49.28%
Percent of adults overweight: 27.54%
Percent of adults healthy: 18.84%
Percent of adults underweight: 4.35%
Average sugar: 35.95 g/day
Average cavities: 2.44

Some analysis of those numbers:
182 is our second largest medical fair. 
54 families of 2 or more is second most we've ever had. 
41 families of 3 or more is the highest we have ever had. 
Gender ratio and adult/child ratio are standard for our medical fairs.
Average age is normal. 
*Need to dig more into bmi categories* 
Soda sugar consumption dropped a lot after first fair but has been slowly increasing since then
Cavities increased from our last fair, this is because sdf - which we used to treat cavities a year ago - is wearing off. It worked well, which is showed by a big reduction of cavities at our last fair, but only lasts 6-12 months.